synapse/models/dns.py

```diff
@@ -37,73 +37,163 @@ def getBaseModels():
+                   {'ptype': 'inet:dns:mx', 'doc': 'Consolidated knowledge of a DNS MX record'}, [
+                      ('fqdn', {'ptype': 'inet:fqdn', 'doc': 'The domain queried for its MX record', 'ro': 1}),
+                      ('mx', {'ptype': 'inet:fqdn', 'doc': 'The domain returned in the MX record', 'ro': 1}),
+                      ('seen:min', {'ptype': 'time:min',
+                                      'doc': 'The earliest observed time for the data in the MX record'}),
+                      ('seen:max', {'ptype': 'time:max',
+                                      'doc': 'The most recent observed time for the data in the MX record'}),
+                ]),
+
+                #An SOA record may return an fqdn or an email address or both.
+                #Based on third party data it seems that there may be different seen:min / seen:max values for
+                #each, so propose creating individual inet:dns:soa records for fqdn + soafqdn or fqdn + soaemail,
+                #but not a single consolidated record for fqdn + soafqdn + soaemail.
+
+                ('inet:dns:soa',
+                   {'ptype': 'inet:dns:soa', 'doc': 'Consolidated knowledge of a DNS SOA record'}, [
+                      ('fqdn', {'ptype': 'inet:fqdn', 'doc': 'The domain queried for its SOA record', 'ro': 1}),
+                      ('soafqdn', {'ptype': 'inet:fqdn', 'doc': 'The domain returned in the SOA record', 'ro': 1}),
+                      ('soaemail', {'ptype': 'inet:email', 'doc': 'The email address returned in the SOA record', 'ro': 1}),
+                      ('seen:min', {'ptype': 'time:min',
+                                      'doc': 'The earliest observed time for the data in the SOA record'}),
+                      ('seen:max', {'ptype': 'time:max',
+                                      'doc': 'The most recent observed time for the data in the SOA record'}),
+                ]),
+
+                #TXT records may have different formats (freeform, attribute=value, used for special purposes like
+                #Sender Policy Framework, DKIM, etc. Those are not taken into account here except as raw strings.
+
+                ('inet:dns:txt',
+                   {'ptype': 'inet:dns:txt', 'doc': 'Consolidated knowledge of a DNS TXT record'}, [
+                      ('fqdn', {'ptype': 'inet:fqdn', 'doc': 'The domain queried for its TXT record', 'ro': 1}),
+                      ('txt', {'ptype': 'str', 'doc': 'The string returned in the TXT record', 'ro': 1}),
+                      ('seen:min', {'ptype': 'time:min',
+                                      'doc': 'The earliest observed time for the data in the TXT record'}),
+                      ('seen:max', {'ptype': 'time:max',
+                                      'doc': 'The most recent observed time for the data in the TXT record'}),
+                ]),
+
                 ('inet:dns:look', {'ptype': 'inet:dns:look', 'doc': 'Instance knowledge of a DNS record lookup'}, [
```

Comment:
this type feels ripe for a bit of tweaking once we figure out what we wanna do with individual protocol instances.

Reply:
Agreed - inet:dns:look instance records could really be (re)formulated as DNS responses. We can leave this in as an interim thing, or remove the additional record types for now if no one is using / has a burning need for instance knowledge of these alternate query / response types.